mentoring/dashboard.py

```diff
@@ -0,0 +1,373 @@
+                return operators[type(node.op)](eval_(node.operand))
+            elif isinstance(node, ast.Name) and node.id == "x":
+                return x
+            elif isinstance(node, ast.BoolOp):  # Boolean operator: either "and" or "or" with two or more values
+                if type(node.op) == ast.And:
+                    return all(eval_(val) for val in node.values)
+                else:  # Or:
+                    for val in node.values:
+                        result = eval_(val)
+                        if result:
+                            return result
+                    return result  # or returns the final value even if it's falsy
+            elif isinstance(node, ast.Compare):  # A comparison expression, e.g. "3 > 2" or "5 < x < 10"
+                left = eval_(node.left)
+                for comparison_op, right_expr in zip(node.ops, node.comparators):
+                    right = eval_(right_expr)
+                    if not operators[type(comparison_op)](left, right):
+                        return False
+                    left = right
+                return True
+            else:
+                raise TypeError(node)
+
+        if not isinstance(expr, ast.AST):
+            expr = ast.parse(expr, mode='eval').body
+        return eval_(expr)
+
+
+@XBlock.needs("i18n")
+class DashboardBlock(StudioEditableXBlockMixin, XBlock):
+    """
+    A block to summarize self-assessment results.
+    """
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Display name for this module"),
+        scope=Scope.settings,
+        default=_('Self-Assessment Summary'),
+    )
+    mentoring_ids = List(
+        display_name=_("Mentoring Blocks"),
+        help=_(
+            "This should be an ordered list of the url_names of each mentoring block whose multiple choice question "
+            "values are to be shown on this dashboard. The list should be in JSON format. Example: {example_here}"
+        ).format(example_here='["2754b8afc03a439693b9887b6f1d9e36", "215028f7df3d4c68b14fb5fea4da7053"]'),
+        scope=Scope.settings,
+    )
+    color_rules = String(
+        display_name=_("Color Coding Rules"),
+        help=_(
+            "Optional rules to assign colors to possible answer values and average values. "
+            "One rule per line. First matching rule will be used. "
+            "Examples: {examples_here}"
+        ).format(examples_here='"1: red", "0 <= x < 5: blue", "green"'),
+        scope=Scope.content,
+        default="",
+        multiline_editor=True,
+        resettable_editor=False,
+    )
+    visual_rules = String(
+        display_name=_("Visual Representation"),
+        default="",
+        help=_("Optional: Enter the JSON configuration of the visual representation desired (Advanced)."),
+        scope=Scope.content,
+        multiline_editor=True,
+        resettable_editor=False,
+    )
```

Comment:
I'm not sure what to suggest about this, but I discovered the images didn't work when reimporting. When I looked at the configuration name, the reason was obvious-- your course name is different than mine. Is there some way to reference course assets without needing the fully qualified URL with course name?